src/gc.c

```diff
@@ -833,6 +833,7 @@ void jl_gc_track_malloced_array(jl_ptls_t ptls, jl_array_t *a)
         ptls->heap.mafreelist = ma->next;
     }
     ma->a = a;
+    ma->how = how;
```

Comment:
Well, we already have an extra bit inside the array so we might as well keep it there :)